dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -497,6 +497,12 @@ private ContextVariables GetNextStepVariables(ContextVariables variables, Plan s
 
         foreach (var item in step.Parameters)
         {
+            // Don't overwrite variable values that are already set except for INPUT
+            if (!item.Key.Equals("INPUT", StringComparison.OrdinalIgnoreCase) && stepVariables.Get(item.Key, out var val))
```

Comment:
Suggest making `ContextVariable.MainKey` public and use it across the solution...


[SKFunction.cs](https://github.com/microsoft/semantic-kernel/blob/f1448c9a0759947256af7f4c6b940c9eab06ea91/dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs#L552)


[SKFunctionInputAttribute](https://github.com/microsoft/semantic-kernel/blob/f1448c9a0759947256af7f4c6b940c9eab06ea91/dotnet/src/SemanticKernel.Abstractions/SkillDefinition/SKFunctionInputAttribute.cs#L51)

Reply:
Will update contextvariables and plan for now, want to keep this scoped.

Reply:
Especially since the casing is different for reasons.